GSoC_Proposal_Automated_Change_Monitoring_ML_NLP.md

```diff
@@ -0,0 +1,192 @@
+
+**Slack handle:** @chaitanyabapat
+
+**Email:** chai.bapat@gmail.com
+
+**Other Contact:**  
+
+**Twitter** : [@ChaiBapchya](https://twitter.com/ChaiBapchya)
+
+**LinkedIn** : [/in/chaitanyaprakashbapat](https://www.linkedin.com/in/chaitanyaprakashbapat)
+
+### Project Details
+
+**Project Title:**  
+
+ML+NLP based Automated Change Monitoring
+
+**Project Abstract/Summary:**  
+
+Generating Environmental corpora, classifying change as relevant or not and prioritizing relevant ones using NLP.
+
+**Describe the need your project fills:**  
+
+_Broader Goal_ :- Categorizing and Prioritizing Changes / Diffs
+
+The project will fulfill the following needs - 
+
+1. _Curated Dataset_
+2. _Classifier_
+3. _Priority algorithm_
+
+**Describe how your project meets this need:**  
+
+Major setbacks
+1. Trump Budget announced - Savings of 31% EPA (reduction in fund allocation)
+2. Not a single corpus / dataset on environment can be found online
+3. Absence of an algorithm to track and prioritize changes
+
+Following needs will be met in my project :-
+1. Dataset - 
+
+After intense deliberation with analysts, subject matter experts (domain experts), we would stitch together a corpus of highly influential and important keywords in the domain of environment.
+
+2. Classifier - 
+
+Using ML techniques (classification algorithms), the diff (from PageFreezer,Versionista) will be categorized as to important, relevant or irrelevant ones by making use of the curated dataset.
+
+Output of Classifier will be cross-checked with the outcomes from NLP techniques (LDA and Summarizer)
+
+3. Priority Algorithm - 
+
+As currently, there is no standard way of prioritizing which change is to be given precendence over the others, which change is critical and more important than others - an algorithm would be needed for the same. Such an algorithm would be on the basis of "Confidence / Trust factor" and "Rating". Taking these 2 parameters into consideration, every change will be evaluated according to the algorithm and then prioritized.
+It would be a self-learning, ever-evolving process of curation.
+
+**Milestones/Timeline:**  
+
+## Phase 0 - Groundzero
+
+1. Understanding the problem at hand - Gathering domain specific knowledge (Environment)
+2. Defining, Reading about meaningful change 
+3. Identify characteristics of changes
+4. Reviewing the existing system Architecture - 
+- Groundwork 
+  -  [edgi-govdata-archiving/web-monitoring#22](https://github.com/edgi-govdata-archiving/web-monitoring/issues/22)
+  -  [edgi-govdata-archiving/overview#102](https://github.com/edgi-govdata-archiving/overview/pull/102)
+  -  [Proposed Services](https://github.com/edgi-govdata-archiving/proposed-services)
+
+
+## Phase 1 - Pre-processing
+
+1. Study PageFreezer API (Webmonitoring processing)
+2. Create corpus of environment data - 
+  + Groundwork - [edgi-govdata-archiving/web-monitoring#28]()
+3. Extracting the content of diff (i.e. Remove html part NLTK - clean_html())
```

Comment:
By removing HTML, I mean the element tags. If it is needed, may be I can store the tags somewhere. Basically seperate HTML element tags from the Actual content.